mainapp/src/Admin.tsx

```diff
@@ -1,30 +1,389 @@
+
+    </>)
+};
+
+type UserFormProps = {
+    modal: boolean,
+    updateModel: React.Dispatch<Msg>,
+    setModal: React.Dispatch<React.SetStateAction<boolean>>,
+    setNewUser: React.Dispatch<React.SetStateAction<User>>,
+    user: User,
+    profiles: string[],
+    saveUser: () => void,
+    deletedUser: () => void
+}
+
+const UserForm: React.FC<UserFormProps> = ({ modal, setModal, updateModel, setNewUser, user, profiles, saveUser, deletedUser }) => {
+    return <Modal open={modal}
+        onClose={restoreUsers(updateModel, setModal)}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description">
+        <Box sx={style}>
+            <Stack spacing={4}>
+                <h2>{`Edit ${user.login}`}</h2>
+                <TextField fullWidth onChange={(event) => setNewUser({ ...user, login: event.target.value })}
+                    //@ts-ignore
+                    value={user.login}
+                    id={`id-login-${user.key}`}
+                    label={"login"}
+                    variant="standard" />
+
+                <TextField fullWidth onChange={(event) => setNewUser({ ...user, password: event.target.value })}
+                    //@ts-ignore
+                    value={user.password}
+                    id={`id-password-${user.key}`}
+                    label={"password"}
+                    variant="standard" />
+                <FormControl>
+                    <InputLabel id="select-groups-label">Groups</InputLabel>
+                    <Select
+                        labelId="select-groups-label"
+                        id="select-groups"
+                        multiple
+                        value={user.groups}
+                        defaultValue={user.groups}
+                        label="select-groups-label"
+                        fullWidth
+                        renderValue={(selected) => typeof selected === 'string' ? selected : selected.join(', ')}
+                        onChange={(event) => setNewUser({ ...user, groups: sanitizeValue(event.target.value) })}
+                    >
+                        {profiles.map(profile => <MenuItem
+                            key={profile}
+                            value={profile}
+
+                        >
+                            {profile}
+                        </MenuItem>)}
+                    </Select>
+                </FormControl>
+
+
+
+
+                <Button onClick={saveUser} variant="contained">Save changes</Button>
+                <Button onClick={deletedUser} variant="contained" color="error">Delete User</Button>
+
+            </Stack>
+        </Box>
+    </Modal >;
+}
+
+function deleteUser(users: User[], user: User, updateModel: React.Dispatch<Msg>, setModal: React.Dispatch<React.SetStateAction<boolean>> = () => console.log("modal closed")) {
+    const updatedUsers = users.filter(prevUsers => prevUsers.key !== user.key);
+    console.log(`user ${user.key} deleted`)
```

Comment:
is this some debus console.log ? Maybe we can remove them ?

Reply:
It had a debugging purpose, it's removed :)